### PR TITLE
feat: disambiguate assemblyscript

### LINKED ISSUE #402: Move asc files to common/contracts/
We currently have AssemblyScript kruft in generated Rust projects:

* `asconfig.js`
* `as-pect.config.js`

This is confusing for people building Rust projects

It's also confusing for people building with AssemblyScript, since these files are in the root of the project directory, rather than being nested within the `assembly` folder where they belong

### PR DESCRIPTION
**Pull request by chadoh**
Fixes #402

* Move assemblyscript smart contract into own subdirectory,
  disambiguating frontend and smart contract dependencies
* Remove `packagejsons` workaround, which resulted in an almost double
  maintenance burden from Dependabot updates
* Fix generated READMEs, since the Vue and Angular templates referred to
  nonexistent files
* Add consistent `compile` and `test` scripts to each kind of contract,
  with `--debug` flag
* Generate symlink to most-recently-compiled version to `out/main.wasm`,
  rather than copying

### CODE REVIEW

File: templates/vue/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+==================
+
+1. The "backend" code lives in the `/contract` folder. See the README there for
+   more info.
+2. The frontend code lives in the `/src` folder. `/src/main.js` is a great
+   place to start exploring.
+3. Tests: there are different kinds of tests for the frontend and the smart
+   contract. See `contract/README` for info about how it's tested. The frontend
+   code gets tested with [jest]. You can run both of these at once with `npm
+   run test`.
```
Comment (chadoh) [inline templates/vue/README.md:32]: @luciotato does this look correct to you? Is there anything you would like to update or add to the NEAR+Vue-specific README?

File: templates/angular/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+1. The "backend" code lives in the `/contract` folder. See the README there for
+   more info.
+2. The frontend code lives in the `/src` folder. `/src/main.ts` is a great
+   place to start exploring.
+   can learn how the frontend connects to the NEAR blockchain.
+3. Tests: there are different kinds of tests for the frontend and the smart
+   contract. See `contract/README` for info about how it's tested. The frontend
+   code gets tested with [jest]. You can run both of these at once with `npm
+   run test`.
```
Comment (chadoh) [inline templates/angular/README.md:33]: @faragly does this look correct to you? Is there anything you would want to add to a NEAR+Angular-specific README?

File: templates/vue/README.md (hunk 1 above)
Comment (luciotato) [reply templates/vue/README.md:32]: looks good to me

File: templates/angular/README.md (hunk 2 above)
Comment (faragly) [reply templates/angular/README.md:33]: this looks good, one note - angular uses a bunch of karma + jasmine for testing

Comment (chadoh) [reply templates/angular/README.md:33]: Updated, thanks!

**Pull request merged**